### PR TITLE
Bug 1131497 - Converted build test scripts to run on node.js

### PR DESCRIPTION
**Pull request by scottwu**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9120f7f12be8bf52b4c8e967479754ae53fe6ab8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=be1c101fc3b1d185e5243251bd1d13a63d0939ea

File: apps/bookmark/test/build/integration/bookmark_test.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 /* jshint node: true, mocha: true */
 /* global suiteSetup */
 
-var helper = require('helper');
+var helper = require('../../../../../build/test/integration/helper.js');
```
Comment (rickychien) [inline apps/bookmark/test/build/integration/bookmark_test.js]: Let's replace that with NODE_PATH=build/test/integration, so that we can get rid of modifying a bunch of files.

File: Makefile [hunk 2]
```diff
@@ -1123,10 +1190,10 @@ really-clean: clean
 	test -d .git && cp tools/pre-commit .git/hooks/pre-commit && chmod +x .git/hooks/pre-commit || true
 
 build-test-unit: b2g_sdk $(NPM_INSTALLED_PROGRAMS)
-	@$(call $(BUILD_RUNNER),build-test,TEST_TYPE=unit REPORTER=$(REPORTER) TRY_ENV=$(TRY_ENV) TEST_FILES="$(TEST_FILES)")
+	@$(call run-node-command,build-test,TEST_TYPE=unit REPORTER=$(REPORTER) TRY_ENV=$(TRY_ENV) TEST_FILES="$(TEST_FILES)")
```
Comment (rickychien) [inline Makefile]: Add ```NODE_PATH=build/test/integration``` at the end to provide additional searching path for node's module loader.

File: Makefile [hunk 3]
```diff
@@ -1123,10 +1190,10 @@ really-clean: clean
...
 
 build-test-integration: b2g_sdk $(NPM_INSTALLED_PROGRAMS)
-	@$(call $(BUILD_RUNNER),build-test,TEST_TYPE=integration REPORTER=$(REPORTER) TRY_ENV=$(TRY_ENV) TEST_FILES="$(TEST_FILES)")
+	@$(call run-node-command,build-test,TEST_TYPE=integration REPORTER=$(REPORTER) TRY_ENV=$(TRY_ENV) TEST_FILES="$(TEST_FILES)")
```
Comment (rickychien) [inline Makefile]: same as above

File: apps/bluetooth/test/build/integration/bluetooth_test.js [hunk 4]
```diff
@@ -3,16 +3,14 @@
 /* jshint node: true, mocha: true */
 /* global suiteSetup */
 
-var helper = require('helper');
+var helper = require('../../../../../build/test/integration/helper.js');
```
Comment (rickychien) [inline apps/bluetooth/test/build/integration/bluetooth_test.js]: Let's use ```NODE_PATH=build/test/integration``` and get rid of modifying for these bunch of files.

File: apps/bluetooth/test/build/integration/bluetooth_test.js [hunk 5]
```diff
@@ -3,16 +3,14 @@
...
 
 suite('Bluetooth tests', function() {
   suiteSetup(helper.cleanupWorkspace);
   teardown(helper.cleanupWorkspace);
-
```
Comment (rickychien) [inline apps/bluetooth/test/build/integration/bluetooth_test.js]: nit: remove this unnecessary change

File: apps/bluetooth/test/build/integration/bluetooth_test.js [hunk 6]
```diff
@@ -3,16 +3,14 @@
...
   test('APP=bluetooth make', function(done) {
     helper.exec('APP=bluetooth make', function(error, stdout, stderr) {
       helper.checkError(error, stdout, stderr);
-
```
Comment (rickychien) [inline apps/bluetooth/test/build/integration/bluetooth_test.js]: nit: remove this unnecessary change

File: build/build-test.js [hunk 7]
```diff
@@ -63,12 +60,12 @@ exports.execute = function(options) {
...
   }
 
-  args = args.concat(files);
-
   // Use Mocha to run test cases.
-  var mocha = new utils.Commander('mocha');
-  mocha.initPath(utils.getFile(options.GAIA_DIR, 'node_modules', '.bin').path);
-  mocha.run(args, function(exitValue) {
-    quit(exitValue);
+  var Mocha = require('mocha');
```
Comment (rickychien) [inline build/build-test.js]: I prefer to do ```require('xxx') at the top of file.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=349af31e143c027ecc611cf335cf24b15654b310

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c17773d5047ea099d331c8b91e228b3cb19b8b29
